clippy_lints/src/methods.rs

```diff
@@ -741,15 +741,18 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
             if let Some(first_arg) = iter_input_pats(&sig.decl, cx.tcx.hir.body(id)).next();
             if let hir::ItemImpl(_, _, _, _, None, ref self_ty, _) = item.node;
             then {
+                if implitem.vis == hir::Visibility::Public ||
+                implitem.vis.is_pub_restricted() {
```

Comment:
pub-restricted methods should be excluded because it is not a public interface. `is_exported` returns false for them, so just fix the `methods.stderr`.